pkg/server/server.go

```
@@ -597,7 +597,10 @@ func NewServer(cfg Config, stopper *stop.Stopper) (serverctl.ServerStartupInterf
 			goschedstats.UnregisterRunnableCountCallback(cbID)
 		}))
 	} else {
-		log.Warning(ctx, "goschedstats not available; admission control will be impaired")
+		if goschedstats.Supported {
+			panic(errors.AssertionFailedf("could not register RegisterRunnableCountCallback"))
```

Curious why `panic` instead of `log.Fatal`?

---

I would change it to `log.Fatal` here, but since the failure is now in `goschedstats`, it stays a `panic`.
<!-- Reviewable comment -OLEs0WS2XQ5QD7-lnpd:b-8w69cx -->
<!-- Sent from Reviewable.io -->